### PR TITLE
Add react guide

### PR DESCRIPTION
**Pull request by simondel**
This guide is meant to help users who want to use Stryker with a React project.

### CODE REVIEW

**Comment by simondel** [comment]
@mthmulders @Archcry @JoshuaKGoldberg Could you provide some feedback on this guide? 

Right now, I have an `npm i -D` command in the guide. We may want to replace this with yarn for the React example but I'm not sure.

File: stryker/react/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+```
+
+## TSX projects
+For projects using TypeScript and TSX you can follow the JSX guide but with a few differences:
+
+Recommended stryker packages: `npm i -D stryker-api stryker stryker-jest-runner stryker-typescript stryker-html-reporter`
+
+Configuration:
+```js
+mutate: ['src/**/*.ts', '!src/**/*@(.test|.spec|Spec).ts'],
```
[the remark this answers is not in the document]
Comment (simondel) [reply stryker/react/README.md]: The spec files for your tsx files would have a .ts extension, right?

[the remark this answers is not in the document]
Comment (ollelauribostrom) [reply stryker/react/README.md]: We are using the .tsx extension for the spec files as well (ex: `Button.spec.tsx`).

**Comment by nicojs** [comment]
I would like to propose a different directory structure. Let's put this guide under `stryker/configuration/react.md`

**Comment by simondel** [comment]
I'll fix the layout of the repo in a separate commit directly to master.

**Comment by mthmulders** [comment]
A bit late to the party, but it looks good to me!

**Comment by nicojs** [comment]
You, my friend, are always right on time for any party.

**Pull request merged**